learntools/machine_learning/ex4.py

```diff
@@ -0,0 +1,78 @@
+from numpy import array
+import pandas as pd
+import sklearn
+from sklearn.tree import DecisionTreeRegressor
+from learntools.core.utils import bind_exercises
+from learntools.core.problem_factories import simple_problem
+from learntools.core.richtext import CodeSolution as CS
+from learntools.core.problem import *
+
+
+class SplitData(CodingProblem):
+    # test are on train_X and val_y. If these are right, others will be right too.
+    _vars = ["train_X", "val_y", "X", "y"]
```

Comment:
Right now running the starter code gives the message:

> Incorrect: You still need to define the following variables: train_X, val_y

This is sort of my fault. I made the assumption that either all variables would be unasssigned to start, or they would all be assigned default values. Anyways, showing an "incorrect" message instead of a "not attempted" one is not a huge deal IMO. But I should consider making this more flexible sometime.